File: packages/ppx/murmur2/test/malformed.js
const murmur2 = require('./murmur2.js');

const buffer_2_bytes = Buffer.from([0xC0]);
const malformed_2_byte_utf8_string = buffer_2_bytes.toString('utf8'); 
const buffer_3_bytes = Buffer.from([0xE0, 0x80]);
const malformed_3_byte_utf8_string = buffer_3_bytes.toString('utf8'); 
const buffer_4_bytes = Buffer.from([0xF0, 0x80]);
const malformed_4_byte_utf8_string = buffer_4_bytes.toString('utf8'); 

process.stdout.write(murmur2(malformed_2_byte_utf8_string) + " (2 bytes) - ");
process.stdout.write(murmur2(malformed_3_byte_utf8_string) + " (3 bytes) - ");
process.stdout.write(murmur2(malformed_4_byte_utf8_string) + " (4 bytes)");

File: packages/ppx/murmur2/test/malformed.ml
let malformed_2_byte_utf8_string =
  String.concat "" [ String.make 1 (Char.chr 0xC0) ]
in
let malformed_3_byte_utf8_string =
  String.concat ""
    [ String.make 1 (Char.chr 0xE0); String.make 1 (Char.chr 0x80) ]
in
let malformed_4_byte_utf8_string =
  String.concat ""
    [ String.make 1 (Char.chr 0xF0); String.make 1 (Char.chr 0x80) ]
in
print_endline
  (Printf.sprintf "%s (2 bytes) - %s (3 bytes) - %s (4 bytes)"
     (Murmur2.default malformed_2_byte_utf8_string)
     (Murmur2.default malformed_3_byte_utf8_string)
     (Murmur2.default malformed_4_byte_utf8_string))
